README.md

```diff
@@ -68,26 +96,46 @@ To run Knot.x you only need Java 8.
 
 To build it you also need Maven.
 
+## Modules
+Knot.x has two modules: **knotx-core** and **knotx-example**.
+
+*Core module* contains Knot.x application without any example data / mock endpoints. It can be deployed as standalone verticle (see http://vertx.io/docs/apidocs/io/vertx/core/Verticle.html). See configuration section to deploy Knot.x core module.
+
+*Example module* contains Knot.x application, example template repositories and mock services. Internally it starts three independent verticles (for Knot.x application, example template repositories and mock services). This module is perfect entry point to Knot.x world. 
+
 ## Building
 
-To build it simply checkout the project, run:
+To build it simply checkout the project, run (from parent directory):
 
 ```
-mvn clean install
+mvn clean package
 ```
-and find the executable jar in the `target` directory.
+and find the executable jar in the `knotx-example/target` directory.
+
+Alternatively, you can just download the most recent `knotx-example-XXX.jar` file.
 
-Alternatively, you can just download the most recent `.jar` file.
+### Executing from Maven
 
-## Executing
+To run it from maven execute the following command from parent directory:
+```
+mvn clean spring-boot:run
```

Comment:
Could we have just `mvn spring-boot:run` here? because in next step i tried to execute fat jar (and it wasn't there :)